spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -137,4 +165,30 @@ public final ColumnarBatch read(ColumnarBatch reuse, int numRowsToRead) {
       return Pair.of(rowIdMapping, currentRowId);
     }
   }
+
+  /**
+   * Reuse the row Id mapping array to filter out equality deleted rows.
+   */
+  private void applyEqDelete(ColumnarBatch batch, int[] posDeleteRowIdMapping, int[] eqDeleteRowIdMapping) {
+    int[] rowIdMapping = posDeleteRowIdMapping == null ? eqDeleteRowIdMapping : posDeleteRowIdMapping;
```

Comment:
It seems like this will only respect one of the two mappings? Or am I wrong there? In general I would try to avoid functions which only use some arguments if possible.

Reply:
Correct. I will move it outside of the method.